x/concentrated-liquidity/position.go

```diff
@@ -95,6 +97,21 @@ func (k Keeper) GetPositionLiquidity(ctx sdk.Context, poolId uint64, owner sdk.A
 	return liquidityStruct.Dec, nil
 }
 
+// GetPositionsLiquidity gets the total liquidity of all positions with varying freeze times for a given owner.
+func (k Keeper) GetPositionsLiquidity(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick, upperTick int64) (sdk.Dec, error) {
+	liquidityArray, err := k.getAllPositionsWithVaryingFreezeTimes(ctx, poolId, owner, lowerTick, upperTick)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	totalLiquidity := sdk.ZeroDec()
+	for _, liquidity := range liquidityArray {
+		totalLiquidity = totalLiquidity.Add(liquidity)
+	}
+
+	return totalLiquidity, nil
+}
```

Comment:
I don't think we should be adding this method to be used in tests only.

Since we treat positions with different IDs as separate, there should be no need for knowing liquidity across all positions within the same tick range.

Additionally, at the moment, it seems to me that the test helper `validatePositionsUpdate` where this function is called should be removed, and `validatePositionUpdate` used instead.